src/ui/public/visualize/loader/embedded_visualize_handler.ts

```diff
@@ -241,22 +246,18 @@ export class EmbeddedVisualizeHandler {
 
   /**
    * renders visualization with provided data
-   * @param visData: visualization data
+   * @param data: visualization data
    */
-  public render = (pipelineResponse: any = {}) => {
+  public render = (data: VisResponseData | null = null) => {
     // TODO: we have this weird situation when we need to render first, and then we call fetch and render ....
     // we need to get rid of that ....
 
-    const renderer = renderFunctionsRegistry.get(get(pipelineResponse, 'as', 'visualization'));
+    const renderer = renderFunctionsRegistry.get(get(data || {}, 'as', 'visualization'));
     if (!renderer) {
       return;
     }
     renderer
-      .render(
-        this.element,
-        pipelineResponse.value || { visType: this.vis.type.name },
-        this.handlers
-      )
+      .render(this.element, get(data, 'value') || { visType: this.vis.type.name }, this.handlers)
```

Comment:
nit: you can use the lodash `get` default value in this case

```suggestion
      .render(this.element, get(data, 'value', { visType: this.vis.type.name }), this.handlers)
```